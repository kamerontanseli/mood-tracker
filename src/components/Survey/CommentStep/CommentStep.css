.CommentStep-title {
  text-align: center;
  margin: 50px 0;
}

.CommentStep-input {
  max-width: 450px;
  margin: 0 auto;
  margin-bottom: 50px;
}

.CommentStep-controls {
  max-width: 200px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
}