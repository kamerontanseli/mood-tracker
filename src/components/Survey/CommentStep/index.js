import "./CommentStep.css";
import React from "react";
import Button from "../../general/Button";
import Panel from "../../general/Panel";
import Header1 from "../../general/Header1";
import TextArea from "../../general/TextArea";

const CommentStep = ({ value, onChange, onNext, onBack }) => (
  <Panel>
    <div className="CommentStep">
      <div className="CommentStep-title">
        <Header1>Anything else to add?</Header1>
      </div>
      <div className="CommentStep-input">
        <TextArea
          cols="30"
          rows="10"
          placeholder="Enter any extra notes about how you feel here..."
          value={value}
          onChange={onChange}
        />
      </div>
      <div className="CommentStep-controls">
        <Button onClick={onBack}>Back</Button>
        <Button onClick={onNext}>Continue</Button>
      </div>
    </div>
  </Panel>
);

export default CommentStep;
