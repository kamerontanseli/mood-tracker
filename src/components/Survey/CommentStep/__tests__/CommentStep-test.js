import React from "react";
import { shallow, mount } from 'enzyme';
import renderer from "react-test-renderer";
import CommentStep from "..";

describe("render", () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <CommentStep value="hello world" />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('should call onChange when textarea is changed', () => {
    const spy = jest.fn()
    const wrapper = mount(<CommentStep value="hello world" onChange={spy} />)
    wrapper.find('textarea').simulate('change')
    expect(spy).toHaveBeenCalled()
  })
  it('should call onBack when [data-test="back"] is clicked', () => {
    const spy = jest.fn()
    const wrapper = mount(<CommentStep value="hello world" onBack={spy} />)
    wrapper.find('[data-test="back"]').first().simulate('click')
    expect(spy).toHaveBeenCalled()
  })
  it('should call onNext when [data-test="next"] is clicked', () => {
    const spy = jest.fn()
    const wrapper = mount(<CommentStep value="hello world" onNext={spy} />)
    wrapper.find('[data-test="next"]').first().simulate('click')
    expect(spy).toHaveBeenCalled()
  })
});
