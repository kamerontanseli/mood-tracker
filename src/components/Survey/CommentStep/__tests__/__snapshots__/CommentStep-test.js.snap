// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<div
  className="Panel"
>
  <div
    className="CommentStep"
  >
    <div
      className="CommentStep-title"
    >
      <h1
        className="Header1"
      >
        Anything else to add?
      </h1>
    </div>
    <div
      className="CommentStep-input"
    >
      <textarea
        className="TextArea"
        cols="30"
        placeholder="Enter any extra notes about how you feel here..."
        rows="10"
        value="hello world"
      />
    </div>
    <div
      className="CommentStep-controls"
    >
      <button
        className="Button"
        data-test="back"
      >
        Back
      </button>
      <button
        className="Button"
        data-test="next"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
